crates/polars-lazy/src/physical_plan/executors/scan/mod.rs

```diff
@@ -101,29 +104,41 @@ pub(crate) struct AnonymousScanExec {
     pub(crate) function: Arc<dyn AnonymousScan>,
     pub(crate) options: AnonymousScanOptions,
     pub(crate) predicate: Option<Arc<dyn PhysicalExpr>>,
+    pub(crate) predicate_has_windows: bool,
 }
 
 impl Executor for AnonymousScanExec {
     fn execute(&mut self, state: &mut ExecutionState) -> PolarsResult<DataFrame> {
-        state.record(
-            || match (self.function.allows_predicate_pushdown(), &self.predicate) {
-                (true, Some(predicate)) => {
+        match (self.function.allows_predicate_pushdown(), &self.predicate) {
```

Comment:
This change is to make the borrow checker happy (due to the mutable borrow of `state`). Is there a simpler approach here?

Reply:
Can take a look later. I think it is fine for this PR.